lib/_http_incoming.js

```diff
@@ -1,7 +1,13 @@
 'use strict';
 
+const binding = process.binding('http_parser');
+const knownHttpHeaders = binding.knownHttpHeaders;
+const lowerCaseHttpHeaderLookup = Object.create(null);
 const util = require('util');
 const Stream = require('stream');
+for (var i = 0; i < knownHttpHeaders.length; i++) {
+  lowerCaseHttpHeaderLookup[knownHttpHeaders[i][1]] = knownHttpHeaders[i][2];
```

Comment:
See [this entry in the V8 wiki](https://github.com/v8/v8/wiki/Design%20Elements#fast-property-access) for more information.

It might be possible to make a constructor that assigns the properties to the instance (`this`) to achieve the same effect, but I cannot say *for sure* whether it has the same performance as the solution I suggested. You'd also need to unset the constructor's `prototype` too. I have used this code generation technique before, so I do know that that works.